src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -347,96 +359,138 @@ public void Dispose()
         GC.SuppressFinalize(this);
     }
 
-#pragma warning disable CA1309 // Use ordinal string comparison. I want to use linguistic comparison.
-    public IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IReadOnlyDictionary<string, string> fieldValues, IDeveloperId developerId)
+    public IAsyncOperation<RepositoriesSearchResult> GetRepositoriesAsync(IReadOnlyDictionary<string, string> fieldValues, IDeveloperId developerId)
     {
-        // First query will find repos only if the user made a PR.
-        var shouldIgnorePRDateOld = _shouldIgnorePRDate;
-        _shouldIgnorePRDate = true;
         return Task.Run(() =>
         {
             try
             {
                 var serverToUse = fieldValues.ContainsKey(_server) ? fieldValues[_server] : string.Empty;
+                _serverToSearch = serverToUse;
+
                 var organizationToUse = fieldValues.ContainsKey(_organization) ? fieldValues[_organization] : string.Empty;
+                _organizationToSearch = organizationToUse;
+
                 var projectToUse = fieldValues.ContainsKey(_project) ? fieldValues[_project] : string.Empty;
 
                 // Get access token for ADO API calls.
                 if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
                 {
                     var exception = new NotSupportedException("The DeveloperId is not valid.");
-                    return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
+                    return new RepositoriesSearchResult(exception, $"{exception.Message} HResult: {exception.HResult}");
                 }
 
-                _azureHierarchy ??= new AzureRepositoryHierarchy(azureDeveloperId);
+                var repos = GetRepos(azureDeveloperId, true);
+                var projects = GetValuesForSearchFieldAsync(fieldValues, _project, developerId).AsTask().Result;

```

Comment:
Since we're doing this in a task could we use async like `Task.Run( async () => { // await code.. });`  so then we wouldn't have to do `.AsTask().Result` or is that not an option here?

Reply:
Changed.